axelor-account/src/main/java/com/axelor/apps/account/web/BudgetController.java

```diff
@@ -45,7 +46,7 @@ public void updateLines(ActionRequest request, ActionResponse response){
 		response.setValue("budgetLineList", budgetLineList);
 	}
 	
-	public void generatePeriods(ActionRequest request, ActionResponse response) {
+	public void generatePeriods(ActionRequest request, ActionResponse response) throws AxelorException {
```

Comment:
Should have a Try catch statement here and store the exception using TraceBackService.